packages/kbn-grid-layout/grid/grid_panel/grid_panel.tsx

```diff
@@ -72,19 +71,20 @@ export const GridPanel = forwardRef<HTMLDivElement, GridPanelProps>(
     /** Set initial styles based on state at mount to prevent styles from "blipping" */
     const initialStyles = useMemo(() => {
       const initialPanel = gridLayoutStateManager.gridLayout$.getValue()[rowIndex].panels[panelId];
+      const { rowHeight } = gridLayoutStateManager.runtimeSettings$.getValue();
       return css`
+        position: relative;
+        height: calc(
+          1px *
+            (
+              ${initialPanel.height} * (${rowHeight} + var(--kbnGridGutterSize)) -
+                var(--kbnGridGutterSize)
+            )
+        );
         grid-column-start: ${initialPanel.column + 1};
         grid-column-end: ${initialPanel.column + 1 + initialPanel.width};
         grid-row-start: ${initialPanel.row + 1};
         grid-row-end: ${initialPanel.row + 1 + initialPanel.height};
-        &.kbnGridPanel--isExpanded {
```

Comment:
All styles related to expanding a panel were moved to the `GridLayout` component so that it is easier to keep track of them.